IHP/IDE/Defaults/TableColumnDefaults.hs

```diff
@@ -0,0 +1,232 @@
+import IHP.IDE.SchemaDesigner.Types
+
+
+{- | Takes a 'Text' value for the name and creates a default column where all values are empty lists, 
+'Nothing' or 'False'; and for 'ColumnType', the default is 'PUUID'.
+
+Add a new field to the 'Column' type in the 
+"IHP.IDE.SchemaDesigner.Types" file and then 
+set its default value here
+
+Defined as:
+
+@
+defColumn :: Column
+defColumn = Column { 
+                 name = ""
+               , columnType = PUUID 
+               , defaultValue = Nothing
+               , notNull = False 
+               , isUnique = False
+               , generator = Nothing
+               }
+@
+
+If you want a different 'PostgresType' you will need to 
+specify with like so by either using a function:
+
+@setColumnType pgt = defColumn {columnType = pgt}@
+
+Or
+
+Just as part of where you're calling it:
+
+@someDefaultColumnType = defColumn {columnType = PDate}@
+-}
+defColumn :: Column
+defColumn = 
+      Column 
+      { name = ""
+      , columnType = PUUID 
+      , defaultValue = Nothing
+      , notNull = False 
+      , isUnique = False
+      , generator = Nothing
+      }
+
+
+{- | Creates an empty table with all values empty. @unlogged@ is set to 'False'.
+
+Defined as such:
+
+@
+emptyTable :: CreateTable
+emptyTable = CreateTable {
+                        name = ""
+                        , columns = []
+                        , primaryKeyConstraint = PrimaryKeyConstraint []
+                        , constraints = []
+                        , unlogged = False
+                        }
+@
+
+-}
+emptyTable :: CreateTable
+emptyTable = CreateTable 
+                  { name = ""
+                  , columns = []
+                  , primaryKeyConstraint = PrimaryKeyConstraint []
+                  , constraints = []
+                  , unlogged = False
+                  }
+
+-- | Takes a name for our table and a list of column and inserts the list
+--   into to our empty table.
+defCreateTable :: Text -> [Column] -> CreateTable
+defCreateTable tablename columns = emptyTable { name = tablename
+                                              , columns = columns }
+
+
+
+
+{- | Creates one default table with a singleton list of one 'setColumn'.
+
+Uses both `defCreateTable` and `setColumn`.
+
+@
+defCreateTableWSetCol :: Text --  The name of the table
+                        -> Text  --  The name of the column
+                        -> PostgresType --  The type of the column
+                        -> CreateTable -- The returned table
+@
+
+-}
+defCreateTableWSetCol :: Text -> Text -> PostgresType -> CreateTable
+defCreateTableWSetCol tablename columnname pgt = defCreateTable tablename (pure $ setColumn columnname pgt)
+
+{- | Same as its progenitor `defCreateTableWSetCol` except it uses `setColumnN`
+-}
+defCreateTableWSetColN :: Text -> Text -> PostgresType -> CreateTable
+defCreateTableWSetColN tablename columnname pgt = defCreateTable tablename (pure $ setColumnN columnname pgt)
```

Comment:
Same with those functions, I think it's more clear to write:

```haskell
emptyTable { name = "users", columns = [ colUUID ] }
```

instead of
```haskell
defCreateTableWSetCol "users" "id" ...
```

Reply:
Same as before, for speed (not needing to write records) and composability (ability to easily use (.) and ($) to change values or combine them with other things). I do understand the concerns for clarity's sake though.

Since there were quite a few examples of where only certain values in certain fields are changed, if you know what the functions do, you can have a quick idea of what the table or column looks like or have a fast way of defining a more complex table as a base for another table by using base functions. 

Again, the option the use the base tables and columns and record syntax is always there and would be functionally identical.

By using a specific function that encapsulates a pattern, I can quickly reason out the structure of the full table if I know what the function does UNLESS I modify it `( )` and adding more records with `{..}` or putting another function ahead of it which modifies the table further.

It really depends on much you want to represent specific patterns of test with functions. Also a lot of these functions can end up being helper functions much larger structures; but these will be made when needed I imagine.